apps/src/templates/feedback/CodeWritten.jsx

```diff
@@ -35,30 +35,32 @@ class CodeWritten extends React.Component {
   };
 
   render() {
+    const {
+      numLinesWritten,
+      totalNumLinesWritten,
+      children,
+      useChallengeStyles
+    } = this.props;
     const lines = (
       <p
         id="num-lines-of-code"
         className="lines-of-code-message"
-        style={
-          this.props.useChallengeStyles ? styles.challengeLineCounts : null
-        }
+        style={useChallengeStyles ? styles.challengeLineCounts : null}
       >
-        {msg.numLinesOfCodeWritten({numLines: this.props.numLinesWritten})}
+        {msg.numLinesOfCodeWritten({numLines: numLinesWritten})}
       </p>
     );
 
     let totalLines;
-    if (this.props.totalNumLinesWritten !== 0) {
+    if (totalNumLinesWritten && totalNumLinesWritten !== numLinesWritten) {
```

Comment:
Ah, ok, this is much clearer, thanks for the refactor!  If you're making another pass, I think a comment here indicating that we don't want to show "total lines" in the UI after the first stage (where totalNumLinesWritten === numLinesWritten) would be helpful, but up to you.